src/odata-to-abstract-sql.ts

```diff
@@ -0,0 +1,1344 @@
+import * as _ from 'lodash';
+import * as memoize from 'memoizee';
+import * as randomstring from 'randomstring';
+import {
+	AbstractSqlQuery,
+	AbstractSqlModel,
+	AbstractSqlTable,
+	Relationship,
+	DurationNode,
+	AbstractSqlType,
+	SelectNode,
+	FromNode,
+	WhereNode,
+	OrderByNode,
+	LimitNode,
+	OffsetNode,
+	NumberTypeNodes,
+	FieldsNode,
+	ValuesNode,
+	ReferencedFieldNode,
+	AliasNode,
+	BooleanTypeNodes,
+	UnionQueryNode,
+	SelectQueryNode,
+	InNode,
+	BindNode,
+	CastNode,
+} from '@resin/abstract-sql-compiler';
+
+export type ResourceNode = ['Resource', string];
+
+declare module '@resin/abstract-sql-compiler' {
+	interface AbstractSqlTable {
+		definition?: Definition;
+	}
+	interface FromNode {
+		1:
+			| SelectQueryNode
+			| UnionQueryNode
+			| TableNode
+			| ResourceNode
+			| AliasNode<SelectQueryNode | UnionQueryNode | TableNode | ResourceNode>;
+	}
+}
+
+export type SupportedMethod =
+	| 'GET'
+	| 'PUT'
+	| 'POST'
+	| 'PATCH'
+	| 'MERGE'
+	| 'DELETE'
+	| 'OPTIONS';
+export type ODataQuery = _.Dictionary<any>;
+export interface ODataBinds extends Array<any> {
+	[key: string]: any;
+}
+
+export interface Definition {
+	extraBinds: ODataBinds;
+	abstractSqlQuery: SelectQueryNode | UnionQueryNode | ResourceNode;
+}
+
+interface Resource extends AbstractSqlTable {
+	tableAlias?: string;
+	definition?: Definition;
+}
+
+const comparison = {
+	eq: 'Equals' as 'Equals',
```

Comment:
So this makes the type be specifically `'Equals'` instead of `string`, so when you do `comparison.eq` the compiler knows that the type is `'Equals'` instead of just any old string.  This matters for
```typescript
return [
	comparison[type as keyof typeof comparison],
	op1,
	op2,
] as BooleanTypeNodes;
```
below as the first elements of BooleanTypeNodes are required to be specific matching strings, so without the `as 'Equals'` here then that will fail due to potentially being invalid

Reply:
Oh and the reason it's not necessary for `operations` just comes down to the abstract sql not being fully typed in all places, so some places are laxer and just accept `[ string, ...AbstractSqlType[] ]` - which happens to be the case where the `operations` nodes are used (over time abstract-sql-compiler will get specific typings for the entire AST, but for now it doesn't)